gql/transport/requests.py

```diff
@@ -36,7 +36,7 @@ def execute(self, document, variable_values=None, timeout=None):
             data_key: payload
         }
         request = requests.post(self.url, **post_args)
-        request.raise_for_status()
+        # request.raise_for_status()
```

Comment:
I think this should be moved to the assertion below. At least in the case when there is no GraphQL result (the assertion fails), raise an error with `request.raise_for_status()` and then an unspecific "raise HTTPError('no GraphQ result')" to catch the case when there is no GraphQL result but the status is good.

Also, shouldn't it be `response = ` instead of  `request = `?

Reply:
Yeah, wasn't sure about moving it around due to the assertion.

Reply:
What do you think about something like this?

```python
response = requests.post(self.url, **post_args)
try:
    result = request.json()
    if not isinstance(result, dict):
        raise ValueError
except ValueError:
    result = {}
if 'errors' not in result and 'data' not in result:
    request.raise_for_status()
    raise HTTPError("Expected a GraphQL result, but got: {}".format(
        response.content), response=response)
return ExecutionResult(
    errors=result.get('errors'), data=result.get('data'))
```

I think that is much better than using an assertion.

Reply:
Or maybe just

    raise HTTPError("Server did not return a GraphQL result", response=response)

to avoid extremely long error messages (e.g. when you get an HTML page instead of GraphQL), since the response will be an attribute of the exception and response.content can be picked up from there.

Reply:
Could work, as long as it return an error response for the user which does not interfer with GraphQL errors.